### PR TITLE
Improved interface for constraints + Removed reflection code

### PR DESCRIPTION
**Pull request by fweik**
- Constraints return forces instead of adding
- Better interface
- const
- Removed buggy reflection code

### CODE REVIEW

File: src/core/constraints/ShapeBasedConstraint.hpp [hunk 1]
```diff
@@ -43,39 +43,27 @@ class ShapeBasedConstraint : public Constraint {
 
   Shapes::Shape const &shape() const { return *m_shape; }
 
-  ReflectionType const &reflection_type() const;
-
   void reset_force() override { m_local_force = Vector3d{0, 0, 0}; m_outer_normal_force=0.0;}
+
   int &only_positive() { return m_only_positive; }
```
Comment (jonaslandsgesell) [inline src/core/constraints/ShapeBasedConstraint.hpp]: this is better a bool, also penetrable

File: src/core/constraints/ShapeBasedConstraint.cpp [hunk 2]
```diff
@@ -109,33 +71,29 @@ void ShapeBasedConstraint::add_force(Particle *p, Vector3d& folded_pos) {
...
         runtimeErrorMsg() << "Constraint"
                           << " violated by particle " << p->p.identity
                           << " dist " << dist;
-      }
     }
   }
-  for (int j = 0; j < 3; j++) {
-    p->f.f[j] += force[j];
+  
+    m_local_force -= force;
```
Comment (KaiSzuttor) [inline src/core/constraints/ShapeBasedConstraint.cpp:85]: Why are those two lines added?

**Review by KaiSzuttor — APPROVED** [review]

File: src/core/constraints/ShapeBasedConstraint.cpp (hunk 2 above)
Comment (fweik) [reply src/core/constraints/ShapeBasedConstraint.cpp:85]: They are just moved up from the end of the function because of the return.

**Comment by jonaslandsgesell** [comment]
Did you change int ->bool also in the tutorials and examples?

**Review by jonaslandsgesell — APPROVED** [review]

**Pull request merged**